rasterio/plot.py

```diff
@@ -130,17 +134,30 @@ def show(source, with_bounds=True, contour=False, contour_label_kws=None,
     return ax
 
 
-def plotting_extent(source):
+def plotting_extent(source, transform=None):
     """Returns an extent in the format needed
      for matplotlib's imshow (left, right, bottom, top)
      instead of rasterio's bounds (left, bottom, top, right)
 
     Parameters
     ----------
-    source : raster dataset
+    source : raster dataset or array
+    transform: Affine, required if source is array
     """
-    extent = (source.bounds.left, source.bounds.right,
-              source.bounds.bottom, source.bounds.top)
+    try:
```

Comment:
@perrygeo this doesn't feel like a try and ask forgiveness type of operation.  I think I'd prefer

```
if hasattr(source, 'bounds'):
    extent = ...
elif transform is None:
    rase ValueError(...)
else:
    transform = guard_transform(transform)
    ...
```

And instead of looking for `bounds` attribute, would it be better to check instance type?  `isinstance(source, rasterio._io.RasterReader)`

(in fact, if we had something in `dtypes` for rasters like we do for ndarrays - `is_ndarray`   we could use that in several places to consistently check if something is a raster or ndarray).